### PR TITLE
Make operators in the specification private protected to allow inheritance.

### PR DESCRIPTION
**Pull request by dcook-net**
Resolves https://github.com/gregsdennis/json-everything/issues/368

### CODE REVIEW

File: JsonLogic/Rules/BooleanCastRule.cs [hunk 1]
```diff
@@ -30,8 +37,6 @@ internal BooleanCastRule(Rule value)
 	/// <returns>The result of the rule.</returns>
 	public override JsonNode? Apply(JsonNode? data, JsonNode? contextData = null)
 	{
-		var value = Value.Apply(data, contextData);
```
Comment (dcook-net) [inline JsonLogic/Rules/BooleanCastRule.cs:33]: This wasn't being used.

Comment (gregsdennis) [reply JsonLogic/Rules/BooleanCastRule.cs:33]: Oh, I probably had that in there as a debug so I could see what the result was.

**Comment by gregsdennis** [comment]
I really need to fix that labeler build.  It's failing because of permissions across forks.

**Comment by gregsdennis** [comment]
@dcook-net can you try rebasing on my `logic/rule-accessibility` branch please?  I don't expect there to be any conflicts.

**Comment by dcook-net** [comment]
@gregsdennis 👍🏻 will do this today.

**Comment by gregsdennis** [comment]
Thanks for this.  I'll merge it into the branch and tweak a few things.  When I merge that into `master` it'll publish.

**Pull request merged**